docs_app/content/tutorial/7-flattening-operations.md

```diff
@@ -0,0 +1,132 @@
+3. Do we really want to stop pushing values to the producer *just* because the outer producer has completed? What if the inner producers aren't complete yet? We won't get all of the values!
+
+These are all important matters, and RxJS resolves these questions with a set of guarantees around the primary flattening operations.
+
+## RxJS Flattening Operator Guarantees
+
+- **All errors are forwarded.** Any error on an outer or inner observable will be pushed do the consumer and stop the subscription.
+- **Source and inner sources must complete.** RxJS flattening operators will return observables whose [subscriptions](GL) will not [complete](GL) until:
+  1. The outer [subscription](GL) is [complete](GL).
+  2. All active inner [subscriptions](GL) have completed*.
+
+    (* there's some slight technical nuance here with the [`switchMap`](API) and [`switchAll`](API) operators.)
+
+## Merge (merge, mergeMap, mergeAll, mergeWith)
+
+Merge is probably the simplest strategy to describe. By default, it will subscribe to every inner observable, as soon as it can, and forward all values from those inner [subscriptions](GL) to the [consumer](GL).
+
+### When to use
+
+- When you don't care about the order of the values or their arrival, you just want them as soon as possible.
+- When you want to kick off [side effects](GL) with observables from notifications. Let's say you want to start an animation with every user action, or that sort of thing, and you want to let those animations run to completion.
+- Merging, in particular, [`mergeMap`](API), is a useful operator right after a [`groupBy`](API) operator. It allows the developer to compose behaviors with the grouped observable emitted by `groupBy`, and then have them rejoin a single observable to be consumed.
+
+### When NOT to use
+
+- When the order of your values from your inner observables matter.
+- HTTP requests where the response matters. (See the previous line item). If your inner observables wrap http requests, then the results of one of those inner subscriptions could arrive after those of a later inner observable. That means that you might end up with out of date data. In particular, you would not want to use this with an HTTP GET. However, if you have an HTTP request that is a pure side effect and you do not care about the response, merge might be a reasonable performance optimization for you.
+
+### Concurrency Limits (Advanced)
+
+Setting a concurrency limit is an advanced feature of some of the merge operators and functions. It allows you to decide how many inner subscriptions to create concurrently, and will queue other inner observables until one of the inner concurrent subscriptions completes, then it will dequeue an inner observable and subscribe to it, pushing its values to the [consumer](GL).
+
+Interestingly, [concat](#concat-concat-concatmap-concatall-concatwith) is a "merge" with a concurrency limit of 1.
+
+## Concat (concat, concatMap, concatAll, concatWith)
```

Comment:
Is `concat` a flattening operator?

Reply:
it is a flattening "operation" encapsulated in a creation function.